### DISCUSSION TITLE
Using Prepare from Automap Finds Non-Existent Table

### DISCUSSION THREAD

**Discussion by RPeterson0487** [sqlalchemy/sqlalchemy#11096] [Usage Questions] [answered]
Hello all,  I've looked through the recent discussions and asked on Gitter to no avail.

I am attempting to use automap's prepare method to reflect a table, but I'm having an issue.  When I run the code, I'm getting a NoSuchTable error for a table that does not show on SHOW FULL TABLES.

Would anyone know where this is coming from if not from that list?

**Comment by RPeterson0487** [comment]
Forgot to add, I am following this example.
https://docs.sqlalchemy.org/en/20/orm/extensions/automap.html

**Comment by zzzeek** [comment]
why does the table not show for SHOW FULL TABLES?

**Reply by RPeterson0487** [reply]
That's a great question.  None of us even know what the table is.  It comes up as an error for a table called Content_Old, but any command on that table fails, including DROP.

**Reply by zzzeek** [reply]
[prepare](https://docs.sqlalchemy.org/en/20/orm/extensions/automap.html#sqlalchemy.ext.automap.AutomapBase.prepare) accepts [reflection options](https://docs.sqlalchemy.org/en/20/core/metadata.html#sqlalchemy.schema.MetaData.reflect), call like this

```py
def _reflect_table(tablename, metadata):
    return tablename != "Content_Old"

mybase.prepare(reflection_options={"only": _reflect_table})
```

**Reply by RPeterson0487** [reply]
I will try that as soon as I get to my computer.  In this case, can I reverse it and only reflect a specified table?  We have hundreds of tables, but will only need to work with one or two at a time.

What I'm wanting to set up is to "convert" a table to a python object, so I can access and change it using something like "Inventory.Location" with Inventory being the object as a given name to access the table Ink_Inventory and Location to be one of the columns.  Is this sort of reflection the best method for this?  The docs seem to show many methods to do this.

**Reply by zzzeek** [reply]
sure, read the docs for [only](https://docs.sqlalchemy.org/en/20/core/metadata.html#sqlalchemy.schema.MetaData.reflect.params.only).     either update that callable to be "tablename in [my list]" or just send the list of names

**Reply by RPeterson0487** [reply]
In the example, it says to use Base.prepare(autoload_with=engine)
Am I replacing the autoload_with with that callable?

**Reply by zzzeek** [reply, marked as answer]
as in my example above, like this:

```py

Base.prepare(autoload_with=engine, reflection_options={"only": ["table1", "table2", "table3"]})
```

**Reply by RPeterson0487** [reply]
You're right, sorry.  Just been a little flustered.  I'm back at my desk and testing now, and so far it looks good.

**Reply by RPeterson0487** [reply]
That seems to get me in the right direction, next is working on how to properly access this information.

Thank you so much for your help.
